Zilon.Core/Zilon.BotEnvironment/AutoplayEngine.cs

```diff
@@ -0,0 +1,166 @@
+            WriteScores(ServiceScope.ServiceProvider, scoreManager, mode, scoreFilePreffix);
+        }
+
+        private static void WriteScores(IServiceProvider serviceFactory, IScoreManager scoreManager, string mode, string scoreFilePreffix)
+        {
+            var summaryText = TextSummaryHelper.CreateTextSummary(scoreManager.Scores);
+
+            Console.WriteLine(summaryText);
+
+            AppendScores(scoreManager, serviceFactory, scoreFilePreffix, mode, summaryText);
+        }
+
+        private static void AppendScores(IScoreManager scoreManager, IServiceProvider serviceFactory, string scoreFilePreffix, string mode, string summary)
+        {
+            var path = SCORE_FILE_PATH;
+            if (!Directory.Exists(path))
+            {
+                Directory.CreateDirectory(path);
+            }
+
+            var botTaskSource = serviceFactory.GetRequiredService<IPluggableActorTaskSource>();
+            var scoreFilePreffixFileName = GetScoreFilePreffix(scoreFilePreffix);
+            var filename = Path.Combine(path, $"{botTaskSource.GetType().FullName}{scoreFilePreffixFileName}.scores");
+            using (var file = new StreamWriter(filename, append: true))
+            {
+                var fragSum = scoreManager.Frags.Sum(x => x.Value);
+                file.WriteLine($"{DateTime.UtcNow}\t{scoreManager.BaseScores}\t{scoreManager.Turns}\t{fragSum}");
+            }
+
+            DatabaseContext.AppendScores(scoreManager, botTaskSource.GetType().FullName, scoreFilePreffix, mode, summary);
+        }
+
+        private void AppendException(Exception exception, string scoreFilePreffix)
+        {
+            var path = SCORE_FILE_PATH;
+            if (!Directory.Exists(path))
+            {
+                Directory.CreateDirectory(path);
+            }
+
+            var botTaskSource = ServiceScope.ServiceProvider.GetRequiredService<IPluggableActorTaskSource>();
+            var scoreFilePreffixFileName = GetScoreFilePreffix(scoreFilePreffix);
+            var filename = Path.Combine(path, $"{botTaskSource.GetType().FullName}{scoreFilePreffixFileName}.exceptions");
+            using (var file = new StreamWriter(filename, append: true))
+            {
+                file.WriteLine(DateTime.UtcNow);
+                file.WriteLine(exception);
+                file.WriteLine();
+            }
+        }
+
+        private static string GetScoreFilePreffix(string scoreFilePreffix)
+        {
+            var scoreFilePreffixFileName = string.Empty;
+            if (!string.IsNullOrWhiteSpace(scoreFilePreffix))
+            {
+                scoreFilePreffixFileName = $"-{scoreFilePreffix}";
+            }
+
+            return scoreFilePreffixFileName;
+        }
+
+        private static void AppendFail(IServiceProvider serviceFactory, string scoreFilePreffix)
+        {
+            Console.WriteLine("[x] Bot task source error limit reached");
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this logging statement.](https://app.codacy.com/manual/kreghek/Zilon_Roguelike/pullRequest?prid=5028977)